compiler/rustc_middle/src/query/mod.rs

```diff
@@ -1100,16 +1100,6 @@ rustc_queries! {
         desc { "destructuring type level constant"}
     }
 
-    /// Tries to destructure an `mir::Const` ADT or array into its variant index
-    /// and its field values. This should only be used for pretty printing.
-    query try_destructure_mir_constant_for_diagnostics(
-        key: (mir::ConstValue<'tcx>, Ty<'tcx>)
```

Comment:
So can we now remove the faulty StableHash implementation(s) that were previously needed to make this work as a query?

Reply:
What "faulty StableHash implementationt(s)" are you thinking about?

Reply:
I'm quoting things @oli-obk said, I don't actually understand the underlying issue here. Also [see Zulip](https://rust-lang.zulipchat.com/#narrow/stream/146212-t-compiler.2Fconst-eval/topic/HashStable.20for.20AllocId.3F).

Reply:
Read the zulip, still don't get it 👍🏻 

Also the "hash eq implies eq" is very scary...